### ISSUE TITLE
AADSTS50000 ERROR and not being askedfor 365 account type or Microsoft account type

### ISSUE THREAD

**Issue by afrancoc2000** [microsoft/azure-spring-boot#663]
## Environment
- Spring boot starter:
  
  - active directory spring boot starter
  
- OS Type: Windows
- Java version: 
  
  - 1.8
  

## Summary
I have both a personal account and a work account with the same email so I expect to be asked for the kind of account to use when I login, but my app looks only for my work account credentials, and when I deploy it to a different tenant, the guest user is actually the one from the personal account, so, I keep getting the AADSTS50000 error.

For what I've read both credentials can be found in https://graph.microsoft.com and I've tried adding https://graph.windows.net and https://outlook.office.com to the scope with no results.

So how can I make my app ask for the account type? and then redirect to https://login.live.com/ if needed?

## Reproduce steps
This is my app code:

SecurityConfig Class

    @EnableWebSecurity
    @EnableGlobalMethodSecurity(prePostEnabled = true)
    public class SecurityConfig extends WebSecurityConfigurerAdapter {

        @Autowired
        private OAuth2UserService<OidcUserRequest, OidcUser> oidcUserService;

        @Override
        protected void configure(HttpSecurity http) throws Exception {

            http
                    .csrf().disable()
                    .authorizeRequests()
                    .antMatchers(HttpMethod.GET,"/api/healthcheck").permitAll()
                    .antMatchers(HttpMethod.OPTIONS).permitAll()
                    .anyRequest().authenticated()
                    .and()
                    .oauth2Login()
                    .userInfoEndpoint()
                    .oidcUserService(oidcUserService)
            ;

            http.logout().logoutRequestMatcher(new AntPathRequestMatcher("/api/logout"))
                    .logoutSuccessUrl("/").deleteCookies("JSESSIONID").invalidateHttpSession(true);
        }
    }    

Application.properties:

    spring.security.oauth2.client.registration.azure.client-id=XXXXXXXXXXX
    spring.security.oauth2.client.registration.azure.client-secret=XXXXXXXXXXX
    spring.security.oauth2.client.registration.azure.client-name=Azure
    spring.security.oauth2.client.registration.azure.provider=azure-oauth-provider
    spring.security.oauth2.client.registration.azure.scope=openid, https://graph.microsoft.com/user.read
    spring.security.oauth2.client.registration.azure.redirect-uri-template={baseUrl}/login/oauth2/code/{registrationId}
    spring.security.oauth2.client.registration.azure.client-authentication-method=post
    spring.security.oauth2.client.registration.azure.authorization-grant-type=authorization_code
    
    spring.security.oauth2.client.provider.azure-oauth-provider.authorization-uri=https://login.microsoftonline.com/{my-tenant}/oauth2/authorize
    spring.security.oauth2.client.provider.azure-oauth-provider.token-uri=https://login.microsoftonline.com/{my-tenant}/oauth2/token
    spring.security.oauth2.client.provider.azure-oauth-provider.user-info-uri=https://login.microsoftonline.com/{my-tenant}/openid/userinfo
    spring.security.oauth2.client.provider.azure-oauth-provider.jwk-set-uri=https://login.microsoftonline.com/{my-tenant}/discovery/keys
    spring.security.oauth2.client.provider.azure-oauth-provider.user-name-attribute=name
    
    azure.activedirectory.tenant-id=common
    azure.activedirectory.active-directory-groups=appUser,appAdmin

A Controller:
    
    @Controller
    public class HomeController {
    
        @PreAuthorize("hasAnyRole('appUser', 'appAdmin')")
        @RequestMapping({"/"})
        public String index() {
            return "index.html";
        }    
    }

## Expected Results
When microsoft checks the email it should ask if the account is the work account or the personal account.

## Actual Results
Microsoft doesn't ask and tries to login with the work account but the user registered is actually the personal account so I get a AADSTS50000 error.

Thanks!

**Comment by zhoufenqin** [comment]
@afrancoc2000 Do you still remember this issue? I don't really understand "a personal account and a work account with the same email", a same account can register for two accounts?

**Comment by afrancoc2000** [comment]
Hi @zhoufenqin I don't remember the issue well, but I work in software development and need to connect sometimes my personal account and the company account for testing. Also, I used to have problems with my enterprise account because there was an issue that didn't allowed me to reset my password.

The password and the AAD issues have been resolved now I think the problem I had, happened while I was trying to use Azure Active Directory V1.0 and everything worked when I moved to AAD V2.0.

Sorry for being vague but I really don't remember much about it.
